### ISSUE TITLE
Script Fields: Boolean Selector

### ISSUE THREAD

**Issue by samuelthng** [home-assistant/frontend#18512]
### Checklist

- [X] I have updated to the latest available Home Assistant version.
- [X] I have cleared the cache of my browser.
- [X] I have tried a different browser to see if it is related to my browser.

### Describe the issue you are experiencing

Using a boolean selector field in a script blueprint with the following configuration:
```yaml
  field_option_one_enabled:
    name: "1️⃣ Option 1"
    description: "Override showing or hiding this option."
    example: !input confirm_enabled
    selector:
      boolean:
```

The following UI can be seen:
![image](https://github.com/home-assistant/frontend/assets/19360561/79eee38c-6f63-4df1-a660-140f0acec395)

When the boolean is checked but value is `false`, the value is not being sent to the script upon service call.
![image](https://github.com/home-assistant/frontend/assets/19360561/1dee8920-aa1f-4861-a650-5edec6a554e6)

However, when you toggle the value to `true` and then back to `false`, the next service call will include the value.

[Example Blueprint](https://github.com/samuelthng/t-house-blueprints/blob/v2.0-beta-2/notifications.yaml)

### Describe the behavior you expected

Value `false` should be set on the variable as long as checkbox on the left is selected, without the need for a toggle.

### Steps to reproduce the issue

1. Create new script with blueprint using boolean selector field
3. Go to Developer Tools → Services
4. Select created script
5. Check the boolean selector and leave value as `false`
6. Call Service

> Note that the service called will not include the boolean selector field in variables.
   `{{ field_option_one_enabled is defined }} # returns false`

7. Change boolean selector value to `true`
8. Change boolean selector value to `false` again

> Note - now value will be defined and return false.
   `{{ field_option_one_enabled is defined }} # returns true`
   `{{ field_option_one_enabled }} # returns false`

### What version of Home Assistant Core has the issue?

2023.11.0

### In which browser are you experiencing the issue with?

Google Chrome Version 118.0.5993.117 (Official Build) (arm64)

### Which operating system are you using to run this browser?

macOS Ventura 13.6.1 (22G313)

### State of relevant entities

```txt
Not state dependent.
```

### Problem-relevant frontend configuration

```yaml
Not configuration dependent (I think - I will gladly provide if required)
```

### Javascript errors shown in your browser console/inspector

```txt
No errors noticed in particular.
```

**Comment by samuelthng** [comment]
Seems it is closed on #13206

**Comment by karwosts** [comment]
Pending fix in #19043
